src/chatd.cpp

```diff
@@ -2088,6 +2109,12 @@ void Chat::handleTruncate(const Message& msg, Idx idx)
                 CALL_DB(setLastReceived, 0);
             }
         }
+
+        if (mClient.isMessageReceivedConfirmationActive() && mLastIdxReceivedFromServer <= idx)
```

Comment:
We try to manage different concepts
    - mLastReceivedIdx and mLastReceivedId are used to save the index and the handle of last message that you have sent to the peer and the peer has confirmed its reception
    - mLastIdxReceivedFromServer and mLastIdReceivedFromServer are used to save the last meesage that you have received from a peer and you have confirmed its reception